crates/p2p/src/commands.rs

```diff
@@ -16,11 +16,8 @@ pub enum Command {
 
     /// Requests some message directly from other operator by public Key.
     RequestMessage {
-        /// Libp2p [`PeerId`] of target peer.
-        ///
-        /// Note: [`PeerId`] can be created from public key of corresponding peer via
-        /// `the_pubkey.to_peer_id()`.
-        peer_id: PeerId,
+        /// Libp2p application [`PublicKey`] of target peer.
+        app_public_key: PublicKey,
```

Comment:
`PublishMessage` needs to be feature-gated to `gossipsub` and `RequestMessage` needs to be feature-gated to `request-response`.

We can choose to not do this in this PR and push it to another PR if you want since a lot of PRs are depending on this one.